src/Controllers/FilesController.js

```diff
@@ -26,7 +26,11 @@ export class FilesController extends AdaptableController {
       contentType = mime.lookup(filename);
     }
 
-    filename = randomHexString(32) + '_' + filename;
+    // increased the random string length to 36 so that it is the same length as an UUID with dashes.
+    // This makes file name property extraction for display in clients such as Parse Dashboard to work correctly.
+    // We can not use an UUID with dashes here because it is already used for files.parse.com hosted files and
+    // expandFilesInObject method below uses this distinction to construct the correct url for the file.
+    filename = randomHexString(36) + '_' + filename;
```

Comment:
if you look at the regex here https://github.com/ParsePlatform/parse-server/blob/e690b73bb5982cb8c8766e92c44e09f6dcd9aa99/src/Controllers/FilesController.js#L9

the separator in the UUID and the rest of the file is a `-` .

IN our case we use a `_` for separating the random string and the filename. So that would still work